weed/storage/idx/walk.go

```diff
@@ -42,6 +42,40 @@ func WalkIndexFile(r io.ReaderAt, fn func(key types.NeedleId, offset types.Offse
 	return e
 }
 
+//copied from WalkIndexFile, just init readerOffset from milestone
+func WalkIndexFileIncrement(r io.ReaderAt, milestone uint64, fn func(key types.NeedleId, offset types.Offset, size types.Size) error) error {
+	var readerOffset = int64(milestone * types.NeedleMapEntrySize)
```

Comment:
the readOffset would not be correct after compaction.

Reply:
Leveldb will be deleted when commiting compact(CommitCompact) before l.load()
Then in function generateLevelDbFile, getMileStone will return 0 because no milestoneKey will be found.
https://github.com/chrislusf/seaweedfs/blob/4820b594a7668db1e31993d473a049c2a48209fb/weed/storage/volume_vacuum.go#L152-L158